packages/tools/client-debug-view/src/components/data-object-views/SharedCounterView.tsx

```diff
@@ -27,7 +43,7 @@ export function SharedCounterView(props: SharedCounterViewProps): React.ReactEle
 
 	React.useEffect(() => {
 		function updateValue(delta: number, newValue: number): void {
-			setValue(newValue);
+			setValue(Math.max(newValue, 0));
```

Comment:
We shouldn't do any modifying of the data we're getting from the SharedCounter. This debug visualizer should be unopinionated about how the SharedCounter's data is being used in the application being debugged,

In fact, in our sandbox app scenario, it should be possible to make this value go negative via the debugger, even though the sandbox app's UI does not allow it. This can be used to test how the application responds to such an occurence.

Reply:
Reverted the changes, and removed `disabled` inside the `input` tag to make sure our debugger goes down to a value below 0.